volatility3/framework/layers/arm.py

```diff
@@ -0,0 +1,915 @@
+        Holds the base address of the translation table for the initial lookup for stage 1 of the translation of an address from the lower VA range in the EL1&0 translation regime, and other information for this translation regime.         [1], see D19.2.155, page 7152
+         [1], see D19.2.152, page 7139
+        """
+
+        ASID = (63, 48)
+        "An ASID for the translation table base address."
+        BADDR = (47, 1)
+        "Translation table base address."
+        CnP = (0, 0)
+        "Common not Private."
+
+    class TTBR1_EL1(Enum):
+        """TTBR1_EL1, Translation Table Base Register 1 (EL1)
+        Holds the base address of the translation table for the initial lookup for stage 1 of the translation of an address from the higher VA range in the EL1&0 stage 1 translation regime, and other information for this translation regime.
+         [1], see D19.2.155, page 7152
+        """
+
+        ASID = (63, 48)
+        "An ASID for the translation table base address."
+        BADDR = (47, 1)
+        "Translation table base address."
+        CnP = (0, 0)
+        "Common not Private."
+
+    class ID_AA64MMFR1_EL1(Enum):
+        """ID_AA64MMFR1_EL1, AArch64 Memory Model Feature Register 1.
+        [1], see D19.2.65, page 6781"""
+
+        HAFDBS = (3, 0)
+        "Hardware updates to Access flag and Dirty state in translation tables."
+
+
+class AArch64RegFieldValues:
+    @classmethod
+    def _table_lookup(
+        cls, value: int, lookup_table: dict, reverse_lookup: bool = False
+    ):
+        if reverse_lookup:
+            lookup_table = {v: k for k, v in lookup_table.items()}
+        if lookup_table.get(value, None) != None:
+            return lookup_table[value]
+        else:
+            raise KeyError(
+                f"Value {value} could not be mapped inside lookup_table : {lookup_table}"
+            )
+
+    @classmethod
+    def _get_feature_HAFDBS(cls, value: int) -> bool:
+        """
+        Hardware updates to Access flag and Dirty state in translation tables.
+         [1], see D19.2.65, page 6784
+        """
+        return value >= 0b10
+
+    @classmethod
+    def _get_ttbr0_el1_granule_size(cls, value: int, reverse_lookup: bool = False):
+        """
+        Granule size for the TTBR0_EL1.
+        """
+        lookup_table = {
+            0b00: 4,  # 4kB
+            0b01: 64,  # 64kB
+            0b10: 16,  # 16kB
+        }
+        return cls._table_lookup(value, lookup_table, reverse_lookup)
+
+    @classmethod
+    def _get_ttbr1_el1_granule_size(
+        cls, value: int, reverse_lookup: bool = False
+    ) -> Optional[int]:
+        """
+        Granule size for the TTBR1_EL1.
+        """
+        lookup_table = {
+            0b01: 16,  # 16kB
+            0b10: 4,  # 4kB
+            0b11: 64,  # 64kB
+        }
+        return cls._table_lookup(value, lookup_table, reverse_lookup)
+
+
+def set_reg_bits(value: int, reg_field: Enum, reg_value: int = 0) -> int:
```

Comment:
Since this seems to relate only to `AArch64RegFieldValues`, could this be made a method of it (a staticmethod or classmethod would be fine), rather than floating off on it's own?